src/openrct2-ui/windows/Map.cpp

```diff
@@ -1097,7 +1097,11 @@ static void window_map_paint_train_overlay(rct_drawpixelinfo* dpi)
         for (auto vehicle_index = train->sprite_index; vehicle_index != SPRITE_INDEX_NULL;
              vehicle_index = vehicle->next_vehicle_on_train)
         {
-            vehicle = GET_VEHICLE(vehicle_index);
+            vehicle = GetEntity<Vehicle>(vehicle_index);
+            if (vehicle == nullptr)
```

Comment:
If such sprite index ends up on the train then the state is already bad. We should have a warning here or it may go unnoticed.

Reply:
I see those sorts of checks as something that should happen at load.

Reply:
I also want to move these sorts of train iterators into the train super class.